### PR TITLE
fix: fixed unit tests and charm code

### PR DESCRIPTION
**Pull request by i-chvets**
Summary of changes:
- Modified test string in unit tests to refreclt updated SDI.
- Modified charm code to use handle errors.
- Updated requirements.

### CODE REVIEW

File: charms/kfp-api/src/charm.py [hunk 1]
```diff
@@ -309,9 +310,11 @@ def _get_interfaces(self):
         try:
             interfaces = get_interfaces(self)
         except NoVersionsListed as err:
-            raise ErrorWithStatus(str(err), WaitingStatus)
+            raise ErrorWithStatus((err), WaitingStatus)
```
Comment (beliaev-maksim) [inline charms/kfp-api/src/charm.py:313]: should it be string or should we remove () ?

Comment (i-chvets) [reply charms/kfp-api/src/charm.py:313]: Pushed. Let's keep it he way it was.

File: charms/kfp-api/tests/unit/test_operator.py [hunk 2]
```diff
@@ -248,7 +252,7 @@ def test_kfp_viz_relation_missing(self, harness: Harness):
                 {"_supported_versions": "- v1", "data": yaml.dump({})},
                 None,
                 pytest.raises(ErrorWithStatus),
-                BlockedStatus("Found empty relation data for kfp-viz"),
+                WaitingStatus("Waiting for kfp-viz relation data"),
```
Comment (beliaev-maksim) [inline charms/kfp-api/tests/unit/test_operator.py:255]: a bit strange, why did outcome change ?

**Review by beliaev-maksim — APPROVED** [review]
LGTM
maybe only this minor from comment https://github.com/canonical/kfp-operators/pull/203/files#r1172852344

**Review by beliaev-maksim — APPROVED** [review]

**Pull request merged**